src/System.Net.HttpListener/src/System/Net/Windows/RequestContextBase.cs

```diff
@@ -116,7 +125,19 @@ protected void UnsetBlob()
 
         protected void SetBuffer(int size)
         {
-            _backingBuffer = size == 0 ? null : new byte[size];
+            if (_backingBuffer != IntPtr.Zero)
+            {
+                Marshal.FreeHGlobal(_backingBuffer);
+            }
+
+            _backingBuffer = size == 0 ? IntPtr.Zero : Marshal.AllocHGlobal(size);
+            _backingBufferLength = size;
+
+            // Zero out the contents of the buffer.
+            for(int i = 0; i < size; ++i)
+            {
+                Marshal.WriteByte(_backingBuffer + i, 0);
```

Comment:
`Marshal.WriteByte` is expensive API to call for something like this.

Instead, cast the `_backingBuffer` to `(byte*)`: `for (int i = 0; i < size; i++) ((byte*)_backingBuffer)[i] = 0;`

Reply:
Will do. I'm glad to know there's a better way to do that.